push-build.sh

```diff
@@ -117,34 +120,34 @@ RELEASE_BUCKET=${FLAGS_bucket:-"kubernetes-release-dev"}
 [[ $KUBE_GCS_UPDATE_LATEST == "n" ]] && FLAGS_noupdatelatest=1
 
 # Default to kubernetes
-: ${FLAGS_release_kind:="kubernetes"}
+: "${FLAGS_release_kind:="kubernetes"}"
 
 # This will canonicalize the path
 KUBE_ROOT=$(pwd -P)
 
 USE_BAZEL=false
-if release::was_built_with_bazel $KUBE_ROOT $FLAGS_release_kind; then
+if release::was_built_with_bazel "$KUBE_ROOT" $FLAGS_release_kind; then
   USE_BAZEL=true
   bazel build //:version
-  LATEST=$(cat $KUBE_ROOT/bazel-genfiles/version)
+  LATEST=$(cat "$KUBE_ROOT"/bazel-genfiles/version)
 else
-  LATEST=$(tar -O -xzf $KUBE_ROOT/_output/release-tars/$FLAGS_release_kind.tar.gz $FLAGS_release_kind/version)
+  LATEST=$(tar -O -xzf "$KUBE_ROOT"/_output/release-tars/$FLAGS_release_kind.tar.gz $FLAGS_release_kind/version)
 fi
-
 if [[ "$LATEST" =~ (${VER_REGEX[release]}(\.${VER_REGEX[build]})?(-dirty)?) ]]; then
+  # shellcheck disable=SC2154
```

Comment:
Can we explain why the disable is needed? Taking a guess:
```suggestion
  # FLAGS_ vars are set dynamically by common::namevalue(); shellcheck can't see their assignments, and we are explicitly checking if they exist
  # shellcheck disable=SC2154
```

Reply:
Your guess is correct :)

Reply:
Will explain in a comment